platforms/documentation/docs/src/docs/userguide/best-practices/best_practices_general.adoc

```diff
@@ -175,3 +175,199 @@ include::sample[dir="snippets/bestPractices/avoidInternal/groovy",files="build.g
 === Tags
 
 <<tags_reference.adoc#tag:upgrades,`#upgrades`>>
+
+[[modularize_builds]]
+== Modularize Your Builds
+:keywords: structuring-builds
+
+Modularize your builds by creating subprojects.
+
+=== Explanation
+
+Splitting your build's source into multiple modules (Gradle projects) is key to taking advantage of Gradle's automatic work avoidance and parallelization features.
+After making a change to a source file, Gradle only needs to recompile affected projects.
+If all your source lives in a single project, Gradle can't skip any recompilations and won't be able to run tasks in parallel.
+You'll likely find other performance benefits of splitting source between projects due to minimizing the compilation classpath of each subproject to only the required dependencies, only running annotation processors on the source files that need them, and so on.
+
+It makes sense to split your build's source into multiple projects _early_.
+Don't wait until you hit some arbitrary number of source files or classes to do this, instead structure your build into multiple projects from the start using whatever natural boundaries exist in your codebase.
+This helps you plan for future growth as your build matures, and improves comprehensibility as each project can be described independently.
+
+Exactly how to split your source in a build varies.
+Common boundaries that make cohesive projects can include horizontal layers (e.g. API vs. implementation, front vs. back end, core business logic vs. UI), vertical slices (e.g. feature modules each containing UI + business logic), inputs to source generation vs. their consumers, or just closely related classes.
```

Comment:
It would be nice to make this a bullet point since this is already a lot of text:
```suggestion
Common boundaries that make cohesive projects can include horizontal layers such as:

- API vs. Implementation
- Front-end vs. Back-end
- Core business logic vs. UI
- Vertical slices (e.g., feature modules each containing UI + business logic)
- Inputs to source generation vs. their consumers
- Or simply closely related classes.
```